python/ray/includes/common.pxi

```diff
@@ -22,12 +22,34 @@ cdef class GcsClientOptions:
 
     @classmethod
     def from_gcs_address(cls, gcs_address):
+        """
+        Create a GcsClientOption that has no cluster_id, and fetches from GCS.
+        """
+        cdef CClusterID c_cluster_id = CClusterID.Nil()
         self = GcsClientOptions()
         try:
             ip, port = gcs_address.split(":", 2)
             port = int(port)
             self.inner.reset(
-                new CGcsClientOptions(ip, port))
+                new CGcsClientOptions(ip, port, CClusterID.Nil(), True, True))
+        except Exception:
+            raise ValueError(f"Invalid gcs_address: {gcs_address}")
+        return self
+
+    @classmethod
+    def create(cls, gcs_address, cluster_id_hex):
```

Comment:
Let's expose `allow_nil` and `fetch_if_nil` here as well since in some python code, cluster_id is guaranteed to be not nil

Reply:
I think eventually we will remove any allow_nil instances (other than the one in ray driver) so it may be an overkill to bind the whole checkings here...

Reply:
I think it's still clear to explicitly set whether a cluster id can be nil or not. It makes the code easier to understand (like I know whether I need to worry about a potential blocking rpc call).